fsthttp/backend.go

```diff
@@ -47,6 +83,33 @@ func (b *Backend) Target() string {
 	return b.target
 }
 
+// Exists returns whether the given backend exists.
+func (b *Backend) Exists() (bool, error) {
+	return fastly.BackendExists(b.name)
+}
+
+// IsHealthy dynamically checks if the backend is available to serve requests.
+func (b *Backend) IsHealthy() (bool, error) {
+	return fastly.BackendIsHealthy(b.name)
+}
+
+// IsDynamic returns whether the backend is dynamic.
+func (b *Backend) IsDynamic() (bool, error) {
+	return fastly.BackendIsDynamic(b.name)
+}
+
+func (b *Backend) Host() (string, error) {
+	return fastly.BackendGetHost(b.name)
+}
+
+func (b *Backend) Port() (int, error) {
+	return fastly.BackendGetPort(b.name)
+}
+
+func (b *Backend) OverrideHost() (string, error) {
+	return fastly.BackendGetOverrideHost(b.name)
+}
+
```

Comment:
Not a big deal, but what do you think about making these fields rather than methods?  I don't think these can be changed after the fact, and it'd eliminate the need to do error checking by users.  They could be populated in `BackendFromName` and `RegisterDynamicBackend` which need to do error handling anyway.

Reply:
That makes sense.  Just realized I didn't fill in all of the possible query fields here.  I think I might try to add Getters to the BackendOptions type and fully populate that on instantiation.